src/v/features/feature_table.cc

```diff
@@ -724,8 +727,19 @@ void feature_table::set_builtin_trial_license(
 
 void feature_table::revoke_license() { _license = std::nullopt; }
 
-const std::optional<security::license>& feature_table::get_license() const {
-    return _license ? _license : _builtin_trial_license;
+const std::optional<security::license>&
+feature_table::get_license(bool use_bootstrap_license) const {
```

Comment:
thought: there are a lot of `use_bootstrap_license = false` cases, so it could make sense to add a new accessor e.g. `feature_table::get_license_or_bootstrap`. 

less code motion in that case, but the compiler isn't much help.

Reply:
I went with the even more specific `should_sanction` now